apps/src/templates/AchievementDialog.jsx

```diff
@@ -195,26 +205,69 @@ const AchievementDialog = Radium(React.createClass({
         handleClose={this.handleClose.bind(this, !tooManyBlocks)}
         assetUrl={this.props.assetUrl}
       >
-        <div style={styles.checkmarks}>
-          {this.achievementRow(true, locale.puzzleCompleted())}
-          {this.achievementRow(!tooManyBlocks, this.blocksUsedMessage(blockDelta, params))}
-          {this.achievementRow(!tooManyHints, this.hintsMessage(tooManyHints))}
-        </div>
-        {this.props.showStageProgress &&
-          <div style={styles.stageRewards}>
-            <div style={styles.stageRewardsTitle}>
-              {locale.stageRewards()}
-            </div>
-            <div style={styles.progressBackground}>
-              <div
-                style={{
-                  ...styles.progressForeground,
-                  width: this.props.stageProgress * MAX_PROGRESS_WIDTH
-                }}
-              />
-            </div>
-          </div>
-        }
+        <StaggeredMotion
+          defaultStyles={Array(4).fill({ progress: 0 })}
+          styles={prevInterpolatedStyles => prevInterpolatedStyles.map((_, i) => {
+            return i === 0 ?
+              { progress: spring(1, {stiffness: 50, damping: 25}) } :
+              { progress: spring(
+                  prevInterpolatedStyles[i - 1].progress > 0.8 ? 1 : 0,
+                  {stiffness: 30, damping: 25}),
+              };
+          })}
+        >
+          {interpolatingValues => {
+              const interpolatingStyles =
+                interpolatingValues.map(val => ({ opacity: val.progress }));
+              return (<div>
+                <div style={styles.checkmarks}>
+                  {this.achievementRow(
+                      true,
+                      locale.puzzleCompleted(),
+                      interpolatingStyles[1])}
```

Comment:
Yeah. Not great, but IMO better than using something like a setTimeout to change state.